### PR TITLE
Convert pkg/telemetry and comp/core/telemetry into go modules

### PR DESCRIPTION
**Pull request by dineshg13**
### What does this PR do?

For motivation refer [RFC](https://docs.google.com/document/d/1uKWa9rMNm0U-euEzEJ69KW7Id5u1E4GfYHyzC1SzYoI/edit#heading=h.wnkz5u7vx1fb)

This PR convert `pkg/telemetry` and `comp/core/telemetry` into go modules. 

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by pgimalac — APPROVED** [review]
Can you add both modules to `task/modules.py` in this PR please ?
Can you either add `skip-qa` or QA instructions ? (I'd say `skip-qa`)

**Review by ogaca-dd — APPROVED** [review]

**Review by IvanTopolcic — APPROVED** [review]

File: pkg/serverless/tags/testGetRuntimeRetries/13/environ [hunk 1]
```diff
@@ -0,0 +1 @@
+AWS_EXECUTION_ENV=nodejs14.x
```
Comment (chouquette) [inline pkg/serverless/tags/testGetRuntimeRetries/13/environ:1]: Is this related to the PR?

**Review by chouquette — APPROVED** [review]
LGTM (with a small question about the change in `pkg/serverless/tags/`)

File: pkg/serverless/tags/testGetRuntimeRetries/13/environ (hunk 1 above)
Comment (dineshg13) [reply pkg/serverless/tags/testGetRuntimeRetries/13/environ:1]: No. looks like this file gets generated when we run tests.

Comment (dineshg13) [reply pkg/serverless/tags/testGetRuntimeRetries/13/environ:1]: Will remove
